src/github3/session.py

```diff
@@ -64,17 +64,40 @@ def __call__(self, request):
 
 
 class GitHubSession(requests.Session):
-    """Our slightly specialized Session object."""
+    """Our slightly specialized Session object.
+
+    Normally this is created automatically by
+    :class:`~github3.github.GitHub`.  To use alternate values for
+    network timeouts, this class can be instantiated directly and
+    passed to the GitHub object.  For example:
+
+    .. code-block:: python
+
+       gh = github.GitHub(session=session.GitHubSession(
+           default_connect_timeout=T, default_read_timeout=N))
+
+    :param default_connect_timeout:
+       the number of seconds to wait when establishing a connection to
+       GitHub
+    :type default_connect_timeout:
+       float
+    :param default_read_timeout:
+       the number of seconds to wait for a response from GitHub
+    :type default_read_timeout:
+       float
+    """
 
     auth = None
     __attrs__ = requests.Session.__attrs__ + [
         "base_url",
         "two_factor_auth_cb",
     ]
 
-    def __init__(self):
+    def __init__(self, default_connect_timeout=4, default_read_timeout=1):
```

Comment:
Seen the read timeout hit locally at 1 second for a github enterprise instance but work consistently at 2. Seemed to only hit for POST requests for creating repos, but might also be hit for searches based on how the documentation states that API behaves. See #919 for details in case there are concerns about increasing the timeout.